src/main/java/cart/ui/ControllerExceptionHandler.java

```diff
@@ -2,22 +2,66 @@
 
 import cart.exception.AuthenticationException;
 import cart.exception.CartItemException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 @ControllerAdvice
-public class ControllerExceptionHandler {
+public class ControllerExceptionHandler extends ResponseEntityExceptionHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass().getSimpleName());
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotValidException ex, HttpHeaders headers, HttpStatus status, WebRequest request) {
+        Map<String, Object> body = new HashMap<>();
+
+        body.put("statusCode", HttpStatus.BAD_REQUEST);
+        body.put("timestamp", LocalDateTime.now());
+
+        List<String> errors = ex.getBindingResult().getFieldErrors()
+                .stream()
+                .map(DefaultMessageSourceResolvable::getDefaultMessage)
+                .collect(Collectors.toList());
+        body.put("messages", errors);
 
-    @ExceptionHandler(AuthenticationException.class)
-    public ResponseEntity<Void> handlerAuthenticationException(AuthenticationException e) {
+        return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
```

Comment:
R : 다른 메서드들과 같은 형식을 유지하는게 좋을거 같아 🙂
```suggestion
        return ResponseEntity
                .status(BAD_REQUEST)
                .body(body);
```

Reply:
오..! 꼼꼼하게 못 봤네 🥲 반영해뒀어